src/workers/userUpdatedPlusSubscription.ts

```diff
@@ -0,0 +1,44 @@
+import { ghostUser } from '../common';
+import { TypedWorker } from './worker';
+import { addNewSourceMember } from '../schema/sources';
+import { SourceMemberRoles } from '../roles';
+import { User } from '../entity';
+
+const PLUS_MEMBER_SQUAD_ID = '05862288-bace-4723-9218-d30fab6ae96d';
+const worker: TypedWorker<'user-updated'> = {
+  subscription: 'api.user-updated-plus-subscribed',
+  handler: async (message, con, log) => {
+    const {
+      data: { newProfile: user, user: oldUser },
+    } = message;
+
+    const beforeFlags = JSON.parse(
+      (oldUser.subscriptionFlags as string) || '{}',
+    ) as User['subscriptionFlags'];
+    const afterFlags = JSON.parse(
+      (user.subscriptionFlags as string) || '{}',
+    ) as User['subscriptionFlags'];
+
+    if (
+      user.id === ghostUser.id ||
+      !user.infoConfirmed ||
+      !!beforeFlags?.cycle
+    ) {
+      return;
+    }
+
+    if (afterFlags?.cycle === beforeFlags?.cycle) {
+      return;
+    }
+
+    const member = {
+      sourceId: PLUS_MEMBER_SQUAD_ID,
+      userId: user.id,
+      role: SourceMemberRoles.Member,
+    };
+    await addNewSourceMember(con, member);
```

Comment:
This can fail if they're already a member right?

Reply:
Guess it can yes, I can add a check before since it's worker anyway.
But very edge-casey I guess